pkg/apis/intevents/v1alpha1/brokercell_types.go

```diff
@@ -68,8 +68,29 @@ var (
 	_ duckv1.KRShaped = (*BrokerCell)(nil)
 )
 
+// ComponentParameters specifies scaling and resource parameters to be used
+// by a single component of a BrokerCell.
+type ComponentParameters struct {
+	// MinReplicas specifies the minimum replica count for the component.
+	MinReplicas *int32 `json:"minReplicas,omitempty"`
+
+	// MaxReplicas specifies the maximum replica count for the component.
+	MaxReplicas *int32 `json:"maxReplicas,omitempty"`
+}
+
+// ComponentsParametersSpec specifies separate parameters for each component
+// of a BrokerCell.
+type ComponentsParametersSpec struct {
+	Fanout  ComponentParameters `json:"fanout,omitempty"`
```

Comment:
I think having independent structs for each components is probably more future proof, e.g., 
```
type FanoutParameters struct{
  ComponentParameters
  OtherParameters...
}
```

This way we can add component specific parameters such as the concurrency configuration for the fanout pool.

Reply:
Not sure we want to add that level of configurability at this point. The purpose of this PR is to get the minimum parameters needed to control for the most basic aspect of how these tests consume resources. While it might be worthwhile to have independent structs for each component, this is something that, imo, we should only do in the future if we indeed need more specific parameters. Since this is all an internal api, afaiu, then it should be fine to iteratively change it instead of trying to design for features we are not sure we need. Let me know if you think otherwise, if so, we'll likely have to start a design doc to outline use case (other than the perf benchmark improvements) that this change should cover.

Reply:
SGTM. As you said this is an internal API we should be able to change it the way we want later on.